Mobile Bridge Cleanup

chrismgeorge opened the pull request:
- [x] Enable camera controls on model drag
- [x] Add way to refresh mobile if on different tab
- [x] Cleanup environment image / skybox situation
- [x] Cleanup any situation of URLs with mobile, how they don't update if they become undefined
- [x] Check if timing out is happening on deployed version
- [x] Make AR settings visible before deploy
- [x] Enable AR should be before Default AR mode 
- [x]  change iOS Settings --> To enable AR on iOS, upload:
- [x] Change iOS upload button to usdz / .reality
- [x] Change wording for explanations
- [x] Add message on iOS device to upload a usdz/.reality if one isn't uploaded
- [x] Remove camera button, add REFRSH MOBILE text to button

chrismgeorge commented:
tagging @mounirlamouri for visibility

chrismgeorge commented:
<img width="1344" alt="Screen Shot 2021-01-13 at 3 45 40 PM" src="https://user-images.githubusercontent.com/26069485/104508556-6a413800-55b6-11eb-91cf-3b7928f1879a.png">

W/ Refresh button when refreshable

chrismgeorge commented on packages/space-opera/src/components/mobile_view/reducer.ts:
```diff
@@ -54,4 +55,34 @@ export function arReducer(
+    state.entities.mobile.isRefreshable;
+export const getForcePost = (state: State) => state.entities.mobile.forcePost;
+
+export function mobileReducer(
```
Added to state so that refreshable changes can interact with the displaying the refresh button in the model viewer preview.

chrismgeorge commented:
<img width="1170" alt="Screen Shot 2021-01-13 at 3 54 55 PM" src="https://user-images.githubusercontent.com/26069485/104509455-b2ad2580-55b7-11eb-8225-629c2b470aef.png">

Modal w/ note

elalish commented on packages/space-opera/src/components/mobile_view/components/mobile_expandable_section.ts:
```diff
@@ -105,23 +105,28 @@ export class MobileExpanadableSection extends LitElement {
     </me-checkbox>
+    <me-checkbox
+      id="ar-modes"
+      label="Default AR Modes to Scene Viewer"
```
Maybe just "Mode" instead of "Modes".

elalish commented on packages/space-opera/src/components/mobile_view/components/mobile_modal.ts:
```diff
@@ -58,11 +58,14 @@ export class MobileModal extends ConnectedLitElement {
     </div>
     <canvas id="qr" style="display: block; margin-bottom: 90px;"></canvas>
+    <div class="modal-text">
+      This uses a third-party <a href="https://github.com/nwtgck/piping-server" target="_blank" class="piping-link">piping server</a> to deploy to your mobile device. This server does not store data.
```
Let's also display the URL of the actual piping server we use, in case someone wants to check that it's legit.

chrismgeorge replied:
You're referring to : https://piping.glitch.me/ correct?

elalish replied:
Yes

elalish commented on packages/space-opera/src/components/mobile_view/mobile_view.ts:
```diff
@@ -64,14 +66,12 @@ export class MobileView extends LitElement {
-      snippet.config.environmentImage = this.config.environmentImage;
-    }
+  get needIosSrc(): boolean {
+    return this.sessionOs === 'iOS' && this.iosUrl.length <= 1;
```
Why `<= 1`?

chrismgeorge replied:
just a metric for determining whether or not there exists a url yet by the string,

elalish replied:
I guess I just figured the check would be for `< 1` or `=== 0`. Is there some special 1-character case?

elalish reviewed (approved):
LGTM.

chrismgeorge commented:
The last commit just addresses the feedback from our meeting and the comments you left here. 

I don't think it is necessary to conditionally reveal the refresh button when the initial tab is open. I, personally, like the consistency with the fact that it is always there.

elalish reviewed (approved):
LGTM, thanks.
